contracts/profiles/DNSResolver.sol

```diff
@@ -0,0 +1,142 @@
+pragma solidity ^0.5.0;
+
+import "../ResolverBase.sol";
+import "../util/RRUtils.sol";
+
+contract DNSResolver is ResolverBase {
+    using RRUtils for *;
+    using BytesUtils for bytes;
+
+    bytes4 constant private DNS_RECORD_INTERFACE_ID = 0xa8fa5682;
+
+    // DNSRecordChanged is emitted whenever a given node/name/resource's RRSET is updated.
+    event DNSRecordChanged(bytes32 indexed node, bytes name, uint16 resource, bytes record);
+    // DNSRecordDeleted is emitted whenever a given node/name/resource's RRSET is deleted.
+    event DNSRecordDeleted(bytes32 indexed node, bytes name, uint16 resource);
+    // DNSZoneCleared is emitted whenever a given node's zone information is cleared.
+    event DNSZoneCleared(bytes32 indexed node);
+
+    // Version the mapping for each zone.  This allows users who have lost
+    // track of their entries to effectively delete an entire zone by bumping
+    // the version number.
+    // node => version
+    mapping(bytes32=>uint16) private versions;
+
+    // The records themselves.  Stored as binary RRSETs
+    // node => name => version => resource => data
```

Comment:
How is `name` represented as a uint16, and why is `version` a bytes32?

Reply:
Comment mistake, should be node=>version=>name=>... for the two mappings (that one and the one below it).